### ISSUE TITLE
[azd pipeline provider] Hard to know the valid strings for `--auth-type`

### ISSUE THREAD

**Issue by vhvb1989** [Azure/azure-dev#1488]
Running `azd pipeline config --help` does not show what are valid entries or the auth-type:

```
vivazqu@ubuntu22:~/workspace/build/i$ azdd pipeline config --help
Create and configure your deployment pipeline by using GitHub Actions or Azure Pipelines.

For more information, go to https://aka.ms/azure-dev/pipeline.

Usage:
  azd pipeline config [flags]

Flags:
      --auth-type string        The authentication type used between the pipeline provider and Azure for deployment (Only valid for GitHub provider)
  -e, --environment string      The name of the environment to use.
  -h, --help                    Gets help for config.
      --principal-name string   The name of the service principal to use to grant access to Azure resources as part of the pipeline.
      --principal-role string   The role to assign to the service principal. (default "contributor")
      --provider string         The pipeline provider to use (github for Github Actions and azdo for Azure Pipelines). (default "github")
      --remote-name string      The name of the git remote to configure the pipeline to run on. (default "origin")

Global Flags:
  -C, --cwd string   Sets the current working directory.
      --debug        Enables debugging and diagnostics logging.
      --no-prompt    Accepts the default value instead of prompting, or it fails if there is no default.

Please let us know how we are doing: https://aka.ms/azure-dev/hats
```

The only way of discovering, is by trying some value and looking at the error msg:

<img width="762" alt="image" src="https://user-images.githubusercontent.com/24213737/216733193-a9f4acfe-942a-479e-8577-f9f3477c462a.png">

Ideally, help should tell what are the valid values.

@savannahostrowski @puicchan , please define the working for the help to mention the valid values

**Comment by savannahostrowski** [comment]
@Austinauth Do you think that we can slip this into the current mocks for help improvements? Or should we allocate to this to an upcoming iteration?

**Comment by puicchan** [comment]
Or we can add `. Valid values: federated, client-credentials)` to the help text, i.e.,  
```
      --auth-type string        The authentication type used between the pipeline provider and Azure for deployment (Only valid for GitHub provider. Valid values: federated, client-credentials.)
```

**Comment by savannahostrowski** [comment]
Yeah, that's the most obvious solution. I do want to see what Austin's take is here though because the string gets a bit long + because we have all `help` layouts (see https://github.com/Azure/azure-dev/issues/822)

**Comment by Austinauth** [comment]
Created a quick mock-up, I hadn't done one for `pipeline config`. We can probably include the "_Valid values: federated, client-credentials_" in the description since it's already a longer one. The alternative would be to include it as a second bullet point in the primary description up top. 

Side note, are there any obvious examples for this command we should include?

https://www.figma.com/file/iIRRiGPuKIfaSFfMJXcYle/azd-resource-provisioning?node-id=1691%3A141013&t=LioKI5cfyS2Otg0W-4

**Comment by savannahostrowski** [comment]
I think this mock looks fine to me. I just didn't want to okay it if you hadn't seen it, given all the work you've put into these screens!

**Comment by Austinauth** [comment]
All good, I appreciate that @savannahostrowski! If you think this scenario is likely to confuse users often, I would definitely suggest we take a look at bringing the message into the primary description. 

To be honest I'm still a little a confused about how the command flag works here. Is this command essentially just printing the authentication type based on if you input `federated` or `client-credentials`?

**Comment by savannahostrowski** [comment]
You'd specify like `azd pipeline config --auth-type federated` for example.
